sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -246,7 +276,40 @@ def display_data(self):
     return self.get_all_options(True)
 
   def view_as(self, cls):
+    """Returns a view of current object as provided PipelineOption subclass.
+
+    Example Usage::
+
+      options = PipelineOptions(['--runner', 'Direct', '--streaming'])
+      standard_options = options.view_as(StandardOptions)
+      if standard_options.streaming:
+        # ... start a streaming job ...
+
+    Note that options objects may have multiple views, and modifications
+    of values in any view-object will apply to current object and other
+    view-objects.
+
+    Args:
+      cls: PipelineOptions class or any of its subclasses.
+
+    Returns:
+      An instance of cls that is intitialized using options contained in current
+      object.
+
+    """
     view = cls(self._flags)
+    for option_name in view._visible_option_list():
+      # Initialize values of keys defined by a cls.
+      #
+      # Note that we do initialization only once per key to make sure that
+      # values in _all_options dict are not-recreated with each new view.
+      # This is important to make sure that values of multi-options keys are
+      # backed by the same list across multiple views, and that any overrides of
+      # pipeline options already stored in _all_options are preserved.
+      if option_name not in self._all_options:
+        self._all_options[option_name] = getattr(view._visible_options,
+                                                 option_name)
+    # Note that views will still store _all_options of the source object.
```

Comment:
Would not this result in views that expose attributes that are not meant to be part of that view?

Reply:
No, what is exposed by a view will be defined by `getattr` which calls `_visible_option_list`, which returns `_visible_options`, which are the options recognized by the parser which are populated during  `view = cls(self._flags)`. The view may still store default values defined by previous views, but should not expose them. It would be prudent to add a test for this, looks like there isn't one already.

I think there will be undefined behavior if multiple options define the same flag with different default values, and the flag is not set on the command line.

Reply:
Added a test.

Reply:
Great, thanks!